docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,180 @@
+tags: [Reference, Log Files]
+sidebar_position: 7
+draft: true
+---
+
+# Log file management plugin
+
+Thin-edge provides an operation plugin to fetch log files from the device.
+
+* The log file management feature is provided with a `tedge-log-plugin` which runs as a daemon on thin-edge.
+* The device owner can define the list of log files in the plugin's configuration file named `tedge-log-plugin.toml`.
+* Each entry in the `tedge-log-plugin.toml` file contains a log `type` and a `path` pattern,
+  where the `type` is used to represent the logical group of log files matching the `path` pattern.
+* Upon receiving a log file upload command for a given `type`, 
+  the log files for that type are retrieved using the `path` pattern defined in this `tedge-log-plugin.toml`,
+  matched against the requested time range, search text and maximum line count.
+* The plugin uploads the requested log file to the tedge file transfer repository.
+  Its url is given by the received log upload command.
+* The list of managed log files in `tedge-log-plugin.toml` can be updated both locally as well as from clouds, for instance, by using the configuration management feature.
+* However, the plugin provides no direct connection to clouds, which is the responsibility of another component, i.e. the cloud mapper.
+* The plugin has a dependency on the `tedge.toml` configuration file to get the MQTT hostname, port, and device identifier.
+
+## Installation
+
+As part of this plugin installation:
+* On systemd-enabled devices, the service definition file for this `tedge-log-plugin` daemon is also installed.
+
+Once installed, the `tedge-log-plugin` runs as a daemon on the device, listening to log update commands on the [`<root>/<identifier>/cmd/log_upload/+` MQTT topic](mqtt-topic-design.md).
+
+## Configuration
+
+The `tedge-log-plugin` configuration is stored by default under `/etc/tedge/plugins/tedge-log-plugin.toml`. If it does not exist on startup, the plugin creates the file with example contents.
+
+This [TOML](https://toml.io/en/) file defines the list of log files that should be managed by the plugin.
+The paths to these files can be represented using [glob](https://en.wikipedia.org/wiki/Glob_(programming)) patterns.
+The `type` given to these paths are used as the log type associated to a log path.
+
+```toml title="file: /etc/tedge/plugins/tedge-log-plugin.toml"
+files = [
+  { type = "mosquitto", path = '/var/log/mosquitto/mosquitto.log' },
+  { type = "software-management", path = '/var/log/tedge/agent/software-*' },
+  { type = "c8y_CustomOperation", path = '/var/log/tedge/agent/c8y_CustomOperation/*' }
+]
+```
+
+The `tedge-log-plugin` parses this configuration file on startup for all the `type` values specified,
+and sends the supported log types message to the MQTT local broker on the `<root>/<identifier>/cmd/log_upload` topic with a retained flag.
+The `<root>/<identifier>` is defined in `tedge.toml` file as `device.topic`.
```

Comment:
I'd have kept separate tedge config keys for the `root` prefix (te/) and the `device identifier` part(device/<device-id>).

Reply:
Divided to `device.topic.root` and `device.topic.id`. If you have a better proposal of naming, please comment here.

Reply:
Having two independent settings makes it easier to change the `root` prefix without knowing anything about the `device identifier`.

For example, if there is only one setting and the default value is `te/device/main///`, if you want to just change the prefix, then the user needs to check what the default value is first, then modify the value to replace `te` with `mycompany`, e.g. `mycompany/device/main///`.

Where as there are two independent settings, then it allows a user to change just the prefix using the following config (without having to know anything about the identifier):

```toml
device.topic.root = "mycompany"
```

Reply:
> Having two independent settings makes it easier to change the `root` prefix without knowing anything about the `device identifier`.

So let's go with 2 settings